### PR TITLE
Fixes #15469: Verifying signatures correctly in shared-files API

### PR DESCRIPTION
**Pull request by ArmandSy**
https://issues.rudder.io/issues/15469

### CODE REVIEW

File: relay/sources/relayd/src/shared_files.rs [hunk 1]
```diff
@@ -257,6 +258,35 @@ pub fn put_handler(
...
+        &meta.algorithm,
+        &meta.digest,
+    );
+
+    if !same_hash_than_in_nodeslist(
+        get_pubkey(meta_pubkey.to_string()).unwrap(),
+        HashType::from_str(hash_type).unwrap(),
+        &meta.digest,
+    )
+    .unwrap()
```
Comment (amousset) [inline relay/sources/relayd/src/shared_files.rs]: We need to properly handle the error here

File: relay/sources/relayd/Cargo.toml [hunk 2]
```diff
@@ -19,45 +19,46 @@ harness = false
 
 [dependencies]
 #### Good
-serde = { version = "1.0", features = ["derive"] }
-serde_json = "1.0"
-nom = "5.0"
+serde = { version = "1.0.98", features = ["derive"] }
```
Comment (amousset) [inline relay/sources/relayd/Cargo.toml]: Why adding precise versions here? Semver should be enough, and this makes it a bit harder to upgrade direct dependencies, while providing no more control than Cargo.lock.

**Comment by ArmandSy** [comment]
Commit modified

File: relay/sources/relayd/Cargo.toml [hunk 3]
```diff
@@ -23,41 +23,42 @@ serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 nom = "5.0"
 # Could use native lib instead, but maybe not prod-ready
-flate2 = { version = "1.0", features = ["zlib"], default-features = false }
-diesel = { version = "1.4", features = ["postgres", "chrono", "r2d2"] }
+flate2 = { version = "1.0.11", features = ["zlib"], default-features = false }
```
Comment (amousset) [inline relay/sources/relayd/Cargo.toml]: Same for all other dependencies, why adding a precise version (which is already fixed in Cargo.lock)

**Comment by ArmandSy** [comment]
Commit modified

File: relay/sources/relayd/Cargo.toml [hunk 4]
```diff
@@ -38,10 +39,9 @@ tokio-threadpool = "0.1"
 tokio-process = "0.2"
 tokio-io = "0.1"
 # Compile dev and release with trace logs enabled
-log = "0.4"
 tracing = { version = "0.1", features = ["max_level_trace", "release_max_level_trace"] }
-tracing-fmt = { git = "https://github.com/tokio-rs/tracing" }
-tracing-log = { git = "https://github.com/tokio-rs/tracing" }
+tracing-fmt = "0.0.1-alpha.3"
+tracing-log = "0.0.1-alpha.1"
```
Comment (amousset) [inline relay/sources/relayd/Cargo.toml]: This should not be part of this PR, can your rebase it ?

**Comment by ArmandSy** [comment]
Commit modified
